airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/convert/SqlTypeToMssqlType.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.mssql.v2.convert
+
+import java.sql.Types
+
+enum class MssqlType(val sqlType: Int, val sqlStringOverride: String? = null) {
+    TEXT(Types.LONGVARCHAR),
+    BIT(Types.BOOLEAN),
+    DATE(Types.DATE),
+    BIGINT(Types.BIGINT),
+    DECIMAL(Types.DECIMAL, sqlStringOverride = "DECIMAL(18, 8)"),
+    VARCHAR(Types.VARCHAR, sqlStringOverride = "VARCHAR(MAX)"),
+    DATETIMEOFFSET(Types.TIMESTAMP_WITH_TIMEZONE),
+    TIME(Types.TIME),
+    DATETIME(Types.TIMESTAMP);
+
+    val sqlString: String = sqlStringOverride ?: name
+
+    companion object {
+        val fromSqlType: Map<Int, MssqlType> =
+            entries
+                .associateByTo(mutableMapOf()) { it.sqlType }
+                .apply { this[Types.TIME_WITH_TIMEZONE] = DATETIMEOFFSET }
```

Comment:
why do we need this? (probably worth adding a code comment)

Reply:
Comment added, it's a reverse-map collision necessity. This might go away once we collapse SqlTypes into MssqlTypes.